docs/handbooks/language-implementation.md

```diff
@@ -0,0 +1,144 @@
+  artifacts? What should the configuration block look like?
+* What is the standard way to publish packages for the new language?
+  - Is there any requirements (code signature, special metadata, ...) that need
+    to be implemented in order to publish valid packages?
+  - How are dependencies modeled? If semantic versioning  is not the norm, what
+    is the strategy to correctly represent semantic version ranges?
+* What are the toolchain and platform requirements?
+
+## Code Generation
+
+Before investing too much work towards the *host library* implementation, it is
+recommended to write a first version of the code generation for the new language.
+This work happens within the [`jsii-pacmak`] package.
+
+Initially, focus on the API signatures: for example, generate code that throws
+a *not implemented* exception when called. Start by making sure a decent API is
+generated from the [`jsii-calc`] package and its dependencies, and use those to
+implement the tests form the [Standard Compliance Suite].
+
+## Host library
+
+Now that we are generating "empty shell" APIs that represent the necessary
+entities to back the [Standard Compliance Suite] tests, start implementing the
+*host library* and update the code generator to make the tests pass until all
+of them pass. It's possible to start publishing artifacts before all the tests
+pass. As soon as some basic features are working, work on [Building and
+Packaging](#building-and-packaging) can start, so early feedback can be
+gathered.
+
+> :construction: A standard architecture for the *host library* has not been
+> documented yet. Upcoming language implementations should contribute to this
+> process by documenting a general architecture that should be implementable
+> in any programming languages (and thus, asbtracting away language
+> specificities).
+
+## Building and Packaging
+
+The necessary toolchains should be added to he [`jsii/superchain`] Docker image,
+so that the [`jsii-pacmak`] generation can be changed to support building ready
+to publish artifacts instead of just code.
+
+Additionally, new constructs have to be added to [`aws-delivlib`] in order to
+support building and publishing of those new artifacts.
+
+## Documentation
+
+Before releasing the new language support to *Developer Preview*, basic
+documentation needs to be produced to explain how to configure a *jsii* project
+to support the new language, and any peculiarities in working with libraries
+generated by [`jsii-pacmak`] for this language.
+
+## Developer Preview
+
+Once the full [Standard Compliance Suite] passes (possibly with the exception of
+certain fringy features), and the documentation covering all aspects of using
+the language bindings have been produced, the new language can be released to
+*Developer Preview*.
+
+It is recommended that new languages stay in *Developer Preview* for a minimum
+of 4 weeks, or until they have received sufficient usage to have built
```

Comment:
4 weeks feels very low, but ok. Should be at least double that.

Reply:
Replaced the `or until` with `ideally until`.